### PR TITLE
Cleanup/reformatting/php8 warnings/notices, some improvements

### PR DESCRIPTION
**Pull request by Klap-in**

### CODE REVIEW

**Review by mprins — APPROVED** [review]
should fix warnings
```
PHP message: PHP Warning:  Undefined array key "comment" in /dokuwiki/lib/plugins/discussion/action.php on line 222
PHP message: PHP Warning:  Undefined array key "number" in /dokuwiki/lib/plugins/discussion/action.php on line 1434
```

File: action.php [hunk 1]
```diff
@@ -1410,45 +1525,50 @@ protected function _render($raw) {
...
         }
 
-        $comments = unserialize(io_readFile($cfile, false));
+        $data = unserialize(io_readFile($file, false));
+
+        $title = $data['title'] ?? '';
 
-        if ($comments['title']) {
-            $title = hsc($comments['title']);
+        $num = $data['number'];
```
Comment (mprins) [inline action.php:1549]: seeing a warning for this line: `PHP message: PHP Warning:  Undefined array key "number" in /dokuwiki/lib/plugins/discussion/action.php on line 1549`

Comment (Klap-in) [reply action.php:1549]: Do I understand correctly that you got this warning when using this pull request?

Comment (mprins) [reply action.php:1549]: yep; on a page that does not have the discussion form showing, but has parts included through blog plugin using `{{blog>blog?2&firstseconly&nofooter&noeditbtn&nocomments}}`; the blog posts have discussion enabled: https://wild-water.nl/dokuwiki/blog/2015/0415_logboek_een_jaar_wildwatertochten
essentially this page: https://wild-water.nl/dokuwiki/start - but on my laptop

Comment (Klap-in) [reply action.php]: You do not see at other places this warning?
I'm wondering if older version of this plugin might have not set this counter..
This one is fixed.

**Comment by Klap-in** [comment]
> should fix warnings
> 
> ```
> PHP message: PHP Warning:  Undefined array key "comment" in /dokuwiki/lib/plugins/discussion/action.php on line 222
> PHP message: PHP Warning:  Undefined array key "number" in /dokuwiki/lib/plugins/discussion/action.php on line 1434
> ```
You meant that this is already solved by this pull request?

File: action.php (hunk 1 above)
Comment (mprins) [reply action.php]: 👍🏽 I updated the branch, deleted the cache and ran though any pages with the discussion tag an nog longer see this popping up

**Pull request merged**